turbopack/crates/turbopack-ecmascript/src/chunk/item.rs

```diff
@@ -85,41 +85,43 @@ impl EcmascriptChunkItemContent {
         }
         .cell())
     }
+}
 
-    #[turbo_tasks::function]
-    pub async fn module_factory(&self) -> Result<Vc<Code>> {
+impl EcmascriptChunkItemContent {
+    async fn module_factory(self: Vc<Self>) -> Result<ResolvedVc<Code>> {
+        let this = self.await?;
```

Comment:
```suggestion
    async fn module_factory(self) -> Result<ResolvedVc<Code>> {
```

I think it makes sense to define this method on the instance itself, instead of the Vc. The caller should await it.